fs/fs_zenfs.h

```diff
@@ -56,12 +61,23 @@ class Superblock {
     nr_zones_ = zbd->GetNrZones();
 
     strncpy(aux_fs_path_, aux_fs_path.c_str(), sizeof(aux_fs_path_) - 1);
+
+    major_zenfs_version_ =
+        static_cast<uint32_t>(std::stoul(ZENFS_VERSION_MAJOR));
+    minor_zenfs_version_ =
+        static_cast<uint32_t>(std::stoul(ZENFS_VERSION_MINOR));
+    patch_zenfs_version_ =
+        static_cast<uint32_t>(std::stoul(ZENFS_VERSION_PATCH));
+    strncpy(commit_zenfs_version_, std::string(ZENFS_VERSION_COMMIT).c_str(),
+            sizeof(commit_zenfs_version_) - 1);
```

Comment:
I don't think that you can actually be sure that `commit_zenfs_version_` is initialized to zero in all implementations. Which mean you don't have a null character at the end of the destination string. Also, no need to go through std::string as far as I can tell:
```suggestion
    strncpy(commit_zenfs_version_, ZENFS_VERSION_COMMIT,
        sizeof(commit_zenfs_version_));
     commit_zenfs_version_[8] = 0;
```

Reply:
This is a non issue now because `commit_zenfs_version_` is now a `uint32_t` :)